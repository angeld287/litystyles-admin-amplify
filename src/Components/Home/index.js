import React, { Component } from 'react';
import { Table, Container, Row, Col, ButtonGroup, Modal, Form } from 'react-bootstrap';

class Home extends Component {

    render() { 
        return (
            <Container>
                <h3 className="mt-5">Home</h3>
            </Container>
        );
    }
}
 
export default Home;