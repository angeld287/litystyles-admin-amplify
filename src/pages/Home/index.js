import React from 'react'
import CardsList from '../../components/CardList';
import { PAGE_OPTIONS } from '../../context/communContex/commun.data';

const Home = () => {
    return (
        <div>
            <CardsList items={PAGE_OPTIONS} />
        </div>
    );
}

export default Home;