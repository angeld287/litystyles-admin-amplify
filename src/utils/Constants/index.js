const SIZE_1024_768 = { name: "_1024_768", dimensions: { width: 1024, height: 768 } }
const SIZE_480_320 = { name: "_480_320", dimensions: { width: 480, height: 320 } }
const SIZE_320_240 = { name: "_320_240", dimensions: { width: 320, height: 240 } }

export const IMAGES = { SIZE_1024_768, SIZE_480_320, SIZE_320_240 };

