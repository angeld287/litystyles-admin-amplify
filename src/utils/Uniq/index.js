export const uniq = (a, param)=>{
    return a.filter(function(item, pos, array){
		return array.map(function(mapItem){ return mapItem[param]; }).indexOf(item[param]) === pos;
	})
}