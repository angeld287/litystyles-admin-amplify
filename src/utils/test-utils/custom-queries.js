import { queryHelpers, buildQueries } from '@testing-library/react'

// The queryAllByAttribute is a shortcut for attribute-based matchers
// You can also use document.querySelector or a combination of existing
// testing library utilities to find matching nodes for your query
const queryAllByDataCy = (...args) =>
    queryHelpers.queryAllByAttribute('data-cy', ...args)

const queryAllByAriaOwns = (...args) =>
    queryHelpers.queryAllByAttribute('aria-owns', ...args)

const queryAllByClass = (...args) =>
    queryHelpers.queryAllByAttribute('class', ...args)

const queryAllByType = (...args) =>
    queryHelpers.queryAllByAttribute('type', ...args)

const getMultipleError = (c, value) =>
    `Found multiple elements with the attribute: ${value}`
const getMissingError = (c, value) =>
    `Unable to find an element with the attribute: ${value}`

const [
    queryByDataCy,
    getAllByDataCy,
    getByDataCy,
    findAllByDataCy,
    findByDataCy,
] = buildQueries(queryAllByDataCy, getMultipleError, getMissingError)

const [
    queryByClass,
    getAllByClass,
    getByClass,
    findAllByClass,
    findByClass,
] = buildQueries(queryAllByClass, getMultipleError, getMissingError)

const [
    queryByAriaOwns,
    getAllByAriaOwns,
    getByAriaOwns,
    findAllByAriaOwns,
    findByAriaOwns,
] = buildQueries(queryAllByAriaOwns, getMultipleError, getMissingError)

const [
    queryByType,
    getAllByType,
    getByType,
    findAllByType,
    findByType,
] = buildQueries(queryAllByType, getMultipleError, getMissingError)

export {
    queryByDataCy,
    queryAllByDataCy,
    getByDataCy,
    getAllByDataCy,
    findAllByDataCy,
    findByDataCy,
    getAllByAriaOwns,
    queryByAriaOwns,
    getByAriaOwns,
    findAllByAriaOwns,
    findByAriaOwns,
    queryAllByAriaOwns,
    queryByClass,
    getAllByClass,
    getByClass,
    findAllByClass,
    findByClass,
    queryAllByClass,
    queryByType,
    getAllByType,
    getByType,
    findAllByType,
    findByType,
}