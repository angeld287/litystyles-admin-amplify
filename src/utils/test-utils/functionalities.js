import { screen, fireEvent, prettyDOM, waitFor } from '@testing-library/react';

const writeInInputFoundByPlaceHolder = (body, placeHolder, text) => {

    const input = screen.getByPlaceholderText(placeHolder);
    fireEvent.change(input, {
        target: { value: text }
    });
    expect(input.value).toBe(text);
}

const SelectItemInModalFoundByTestId = async (body, dataTestId, option) => {

    // find select by data test id
    let select = screen.getByTestId(dataTestId).querySelector('[class=ant-select-selection-search-input]');

    // Dropdown the select
    await waitFor(() => {
        fireEvent.mouseDown(select);
    });

    // Select the option
    fireEvent.click(screen.getByText(option));

    // Close the select
    fireEvent.click(body);
}

const insertFileInInputFileFoundByTestId = async (body, dataTestId, file) => {
    //Get the input file by test id
    let _file = screen.getByTestId(dataTestId);

    //verify if the input file is empty
    expect(_file.files.length).toBe(0);

    //add file to the input file
    await waitFor(() => {
        fireEvent.change(_file, {
            target: { files: [file] },
        })
    });

    //verify if the input file has the new file
    _file = screen.getByTestId(dataTestId);
    expect(_file.files.length).toBe(1);
}

export {
    writeInInputFoundByPlaceHolder,
    SelectItemInModalFoundByTestId,
    insertFileInInputFileFoundByTestId
}