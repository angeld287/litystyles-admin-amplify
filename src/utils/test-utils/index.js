// test-utils.js
import { render, queries } from '@testing-library/react'
import * as customQueries from './custom-queries'
import * as customFunctionalities from './functionalities';

const customRender = (ui, options) =>
    render(ui, { queries: { ...queries, ...customQueries, ...customFunctionalities }, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }