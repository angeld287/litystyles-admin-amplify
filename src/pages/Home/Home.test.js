import { render } from '../../utils/test-utils';
import Home from './index';

test('Verify correct renderization of Home Component', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
})