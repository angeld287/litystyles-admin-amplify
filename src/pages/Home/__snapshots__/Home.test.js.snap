// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify correct renderization of Home Component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="card-container"
    >
      <div
        class="card"
      >
        <header
          class="card-header header-img"
        >
          <p
            class="header-title"
          >
            $224,900
          </p>
        </header>
        <div
          class="card-body"
        >
          3 Rooms, 2 Baths - House for sale
        </div>
        <div
          class="card-footer"
        >
          <button
            class="btn"
          >
            Buy
          </button>
          <button
            class="btn"
          >
            More info
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <header
          class="card-header header-img"
        >
          <p
            class="header-title"
          >
            $224,900
          </p>
        </header>
        <div
          class="card-body"
        >
          3 Rooms, 2 Baths - House for sale
        </div>
        <div
          class="card-footer"
        >
          <button
            class="btn"
          >
            Buy
          </button>
          <button
            class="btn"
          >
            More info
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <header
          class="card-header header-img"
        >
          <p
            class="header-title"
          >
            $224,900
          </p>
        </header>
        <div
          class="card-body"
        >
          3 Rooms, 2 Baths - House for sale
        </div>
        <div
          class="card-footer"
        >
          <button
            class="btn"
          >
            Buy
          </button>
          <button
            class="btn"
          >
            More info
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
